### PR TITLE
EVAKA-HOTFIX: Fix authorization of global scope actions for scoped roles

### PR DESCRIPTION
**Pull request by Joosakur**

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt [hunk 1]
```diff
@@ -145,7 +145,12 @@ WHERE employee_id = :userId
     }
 
     fun requirePermissionFor(user: AuthenticatedUser, action: Action.Global) {
-        assertGlobalPermission(user, action, permittedRoleActions::globalActions)
+        assertPermission(
+            user = user,
+            getAclRoles = { acl.getRolesForAnyUnit(user).roles },
+            action = action,
+            mapping = permittedRoleActions::globalActions
+        )
```
Comment (akheron) [inline service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt:153]: The downside here is that the SQL query is run even if the action doesn’t depend on any scoped roles.

Comment (akheron) [reply service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt:153]: Actually it seems they aren’t after all 👍

Comment (Joosakur) [reply service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt:153]: Are you sure? It's supposed to be a lambda that is only evaluated when necessary

File: service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt [hunk 2]
```diff
@@ -331,7 +341,7 @@ WHERE employee_id = :userId
         return result
     }
 
-    private inline fun <reified A, reified I> ActionConfig<A>.hasGlobalPermission(
+    private inline fun <reified A, reified I> ActionConfig<A>.hasPermissionThroughGlobalRole(
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt:344]: these functions are new to me and I don't know their purpose so I don't know if i broke them

**Review by Gekkio — APPROVED** [review]

**Pull request merged**